### PR TITLE
Upgrade CoinUtils -- 2.11.11

### PR DESCRIPTION
**Pull request by amontoison**
@odow

### CODE REVIEW

**Review by odow — CHANGES_REQUESTED** [review]
Wait for resolution of https://github.com/JuliaPackaging/Yggdrasil/pull/8066#discussion_r1477815422

**Review by odow — APPROVED** [review]
Okay try this now

**Comment by odow** [comment]
A list of the order is:

[CoinUtils_jll]
[Osi_jll] 
[Clp_jll] 
[Cgl_jll] 
[Cbc_jll] 
[Bonmin_jll] 
[SYMPHONY_jll] 
[ALPS_jll] 
[BiCePS_jll] 
[CHIPPS_BLIS_jll] 
[MibS_jll]

**Comment by ViralBShah** [comment]
I believe you have to merge CoinUtils first, and then use it in the other packages in the dependency chain - and keep doing that repeatedly.

**Comment by amontoison** [comment]
Ipopt_jll is already linked with LBT so we could remove the folder for the artifact IpoptMKL_jll.

Yes, I discussed with Oscar a few days ago and we will start with CoinUtils, OsiUtils and Clp.

**Comment by ViralBShah** [comment]
We can just let IpOptMKL be around, but just not update it going forward.

**Comment by odow** [comment]
> I believe you have to merge CoinUtils first, and then use it in the other packages in the dependency chain - and keep doing that repeatedly.

Yeah, @amontoison has this under control, but they're traveling etc over the next few weeks. There is no rush.

**Comment by imciner2** [comment]
> Ipopt_jll is already linked with LBT so we could remove the folder for the artifact IpoptMKL_jll.

Has anyone tested running the LBT build with MKL selected? We should make sure we don't see the same problem that SLICOT had when using MKL with LBT (e.g. https://github.com/JuliaLinearAlgebra/MKL.jl/issues/138).

**Comment by imciner2** [comment]
> But we compile all COIN-OR packages with LP64 BLAS / LAPACK and not ILP64 BLAS / LAPACK so we can't have this problem.
> 
> I also checked with nm -D libCoinUtils.so.3.11.10 and only two symbols (related to dense LU) are used:

Excellent, then it should be safe to deprecate the old IpOptMKL and just use the main build from now on. IMO, I agree we can leave the build scripts there, but I think we should  add a `DEPRECATED` file to the folder explaining the deprecation and that the normal IpOpt should be used instead.

**Comment by ViralBShah** [comment]
Should we merge this and then merge the Clp PR?

**Comment by odow** [comment]
There's now 2.11.11 so we should probably first update to that 
https://github.com/coin-or/CoinUtils/releases/tag/releases%2F2.11.11

**Comment by amontoison** [comment]
I will update the PR tomorrow. I forgot that I started to work on the LBT support in COIN-OR packages.

File: C/Coin-OR/coin-or-common.jl [hunk 1]
```diff
@@ -45,8 +45,8 @@ Clp_gitsha = "1c2586a08d33ecc59ed67d319c29044802c0866b"
 Osi_version = offset_version(v"0.108.7", v"0.0.0")
 Osi_gitsha = "c67528fc8b2a36dc936b285fe34030439b9147c3"
 
-CoinUtils_version = offset_version(v"2.11.6", v"0.0.0")
-CoinUtils_gitsha = "26e9639ed9897e13e89169870dbe910296a9783b"
+CoinUtils_version = offset_version(v"2.11.10", v"0.0.0")
+CoinUtils_gitsha = "5b98c4bb87ceebedb86c6e921dc4ad0af6166c8e"
```
Comment (odow) [inline C/Coin-OR/coin-or-common.jl:49]:
```suggestion
CoinUtils_gitsha = "15a819b7e8763b2557e4bf440f8cb62ee6734f36"
```

File: C/Coin-OR/coin-or-common.jl [hunk 2]
```diff
@@ -45,8 +45,8 @@ Clp_gitsha = "1c2586a08d33ecc59ed67d319c29044802c0866b"
 Osi_version = offset_version(v"0.108.7", v"0.0.0")
 Osi_gitsha = "c67528fc8b2a36dc936b285fe34030439b9147c3"
 
-CoinUtils_version = offset_version(v"2.11.6", v"0.0.0")
-CoinUtils_gitsha = "26e9639ed9897e13e89169870dbe910296a9783b"
+CoinUtils_version = offset_version(v"2.11.10", v"0.0.0")
```
Comment (odow) [inline C/Coin-OR/coin-or-common.jl:48]:
```suggestion
CoinUtils_version = offset_version(v"2.11.11", v"0.0.0")
```

File: C/Coin-OR/CoinUtils/build_tarballs.jl [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # In addition to coin-or-common.jl, we need to modify this file to trigger a
 # rebuild.
 #
-# Last updated: 2022-10-26
+# Last updated: 2024-02-05
```
Comment (amontoison) [inline C/Coin-OR/CoinUtils/build_tarballs.jl:4]:
```suggestion
# Last updated: 2024-06-19
```

**Pull request closed without merging**